
A user reported DMA spinning closing millions of file descriptors when
attempting to create a mailbox.

diff -wpruN '--exclude=*.orig' a~/local.c a/local.c
--- a~/local.c	1970-01-01 00:00:00
+++ a/local.c	1970-01-01 00:00:00
@@ -75,12 +75,16 @@ create_mbox(const char *name)
 	switch (child) {
 	case 0:
 		/* child */
+#if 0
 		maxfd = sysconf(_SC_OPEN_MAX);
 		if (maxfd == -1)
 			maxfd = 1024;	/* what can we do... */
 
 		for (i = 3; i <= maxfd; ++i)
 			close(i);
+#else
+		closefrom(3);
+#endif
 
 		execl(LIBEXEC_PATH "/dma-mbox-create", "dma-mbox-create", name, NULL);
 		syslog(LOG_ERR, "cannot execute "LIBEXEC_PATH"/dma-mbox-create: %m");
